src/error/status.rs

```diff
@@ -127,6 +181,7 @@ impl Status {
     where
         F: FnOnce() -> T,
     {
+        // FIXME: Is that the best way to handle warnings?
```

Comment:
As the comment suggests, I'm not sure if Rust's error handling type comes well-prepared to model UEFI's warning statuses. A better alternative would probably be to only treat erronerous status codes as errors, and express warnings as part of the _valid result_ data model.

Reply:
Maybe, I don't know what the right solution here is. I just tried to find a way to prevent people from ignoring warnings when they occur. While the standard defined warnings can usually be safely ignored, we don't know what the firmware considers a warning.